python/cudf/cudf/core/reshape.py

```diff
@@ -234,13 +279,29 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
         result_columns = objs[0].columns
         for o in objs[1:]:
             result_columns = result_columns.append(o.columns)
-
-        df.columns = result_columns.unique()
         if ignore_index:
-            df.index = cudf.RangeIndex(len(objs[0]))
-            return df
-        elif not match_index:
+            # with ignore_index the column names change to numbers
+            df.columns = pd.RangeIndex(len(result_columns.unique()))
+        if empty_inner:
+            # if join is inner and it containes an empty df
+            # we return an empty df
+            return df.head(0)
+        if not ignore_index:
+            df.columns = result_columns.unique()
+        if not match_index and sort is not False:
             return df.sort_index()
+        if (
+            sort
+            or join == "inner"
+            or (
+                not isinstance(df.index, cudf.MultiIndex)
+                and df.index.dtype == "int"
```

Comment:
Since the issue(https://github.com/pandas-dev/pandas/issues/37937) is on the pandas side, we shouldn't be trying to mimic the similar behavior in cudf.